src/main/java/cws/k8s/scheduler/memory/WaryPredictor.java

```diff
@@ -0,0 +1,199 @@
+        initialValue = new HashMap<>();
+        ignoreList = new HashMap<>();
+        lowestSuccess = new HashMap<>();
+    }
+
+    @Override
+    public void addObservation(Observation o) {
+        log.debug("WaryPredictor.addObservation({})", o);
+
+        // nextflow will only retry once, so if the task failed, we will add
+        // it to our ignore list, so that it wont fail twice
+        if (!Boolean.TRUE.equals(o.success)) {
+            if (!ignoreList.containsKey(o.task)) {
+                ignoreList.put(o.task, new ArrayList<>());
+            }
+            ignoreList.get(o.task).add(o.taskName);
+        }
+        
+        if (!TaskScaler.checkObservationSanity(o)) {
+            log.warn("dismiss observation {}", o);
+            return;
+        }
+
+        // store initial ramRequest value per task
+        if (!initialValue.containsKey(o.task)) {
+            initialValue.put(o.task, o.getRamRequest());
+        }
+
+        if (!overprovisioning.containsKey(o.task)) {
+            overprovisioning.put(o.task, 1.1);
+        }
+
+        if (!errorCounter.containsKey(o.task)) {
+            errorCounter.put(o.task, 0);
+        }
+
+        if (Boolean.TRUE.equals(o.success)) {
+
+            if (!observations.containsKey(o.task)) {
+                observations.put(o.task, new ArrayList<>());
+            }
+            if (!model.containsKey(o.task)) {
+                model.put(o.task, new SimpleRegression());
+            }
+
+            double x = o.getInputSize();
+            double y = o.getPeakVmem().doubleValue();
+
+            lowestSuccess.put(o.task, BigDecimal.valueOf(y));
+            observations.get(o.task).add(Pair.of(x, y));
+            model.get(o.task).addData(x,y);
+        } else {
+            Integer errors = errorCounter.get(o.task);
+            errorCounter.put(o.task, 1+errors);
+            log.debug("overprovisioning value will increase due to task failure, errors: {}", 1+errors);
+            Double old = overprovisioning.get(o.task);
+            overprovisioning.put(o.task, old+0.05);
+        }
+    }
+
+    @Override
+    public String queryPrediction(Task task) {
+        String taskName = task.getConfig().getTask();
+        log.debug("WaryPredictor.queryPrediction({},{})", taskName, task.getInputSize());
+        
+        // check ignore list first
+        if (ignoreList.containsKey(taskName) && (ignoreList.get(taskName).contains(task.getConfig().getName()))) {
+            log.debug("{} is on the ignore list", task.getConfig().getName());
+            return null;
+        }
+
+        if (!model.containsKey(taskName)) {
+            log.debug("WaryPredictor has no model for {}", taskName);
+            return null;
+        }
+        
+        if (2 < errorCounter.get(taskName)) {
+            log.warn("to many errors for {}, providing initial value", taskName);
+            return initialValue.get(taskName).toPlainString();
+        }
```

Comment:
I think we should have a configurable value for this or at least a constant in this class.
This is very specific to some configurations. where a task could only be repeated two times. 
Also please not use the initialValue but the user defined value of the concrete task, if you had to many trials.
The value could be different for all physical tasks.